src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeAnalyzer.cs

```diff
@@ -0,0 +1,267 @@
+    ///
+    ///   * The method cannot be virtual, abstract, or be an override.
+    ///
+    ///   * The method must be private.
+    ///
+    ///   * The method must not have been assigned to a delegate.
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed partial class UseConcreteTypeAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1859";
+
+        internal static readonly DiagnosticDescriptor UseConcreteTypeForMethodReturn = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeTitle)),
+            CreateLocalizableResourceString(nameof(UseConcreteTypeForMethodReturnMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        internal static readonly DiagnosticDescriptor UseConcreteTypeForParameter = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeTitle)),
+            CreateLocalizableResourceString(nameof(UseConcreteTypeForParameterMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        internal static readonly DiagnosticDescriptor UseConcreteTypeForLocal = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeTitle)),
+            CreateLocalizableResourceString(nameof(UseConcreteTypeForLocalMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        internal static readonly DiagnosticDescriptor UseConcreteTypeForField = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeTitle)),
+            CreateLocalizableResourceString(nameof(UseConcreteTypeForFieldMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(UseConcreteTypeDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(UseConcreteTypeForField);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            if (!Debugger.IsAttached)
```

Comment:
Please remove this - most times it is recommended that you debug via minimal unit tests where concurrent behavior should also be fine. Analyzers should never special case for when running under the debugger or not.

Reply:
Done